### ISSUE TITLE
Write User Requirements

### ISSUE THREAD

**Issue by fahreddinozcan** [bounswe/bounswe2024group5#270]
We've got to write the user requirements.

**Comment by arincdemir** [comment]
<details><summary>1.1.1. Account Requirements</summary><blockquote>
- 1.1.1.1 The account shall have a unique username associated with the user.
- 1.1.1.2 The account shall have a changeable and publicly available name associated with the user.
- 1.1.1.3 The account shall have a changeable and publicly available profile picture associated with the user.
- 1.1.1.4 The account shall have a field for users to specify their English proficiency level (e.g., beginner, intermediate, advanced).
- 1.1.1.5 The account shall track and display the user's accumulated points over time.
- 1.1.1.6 The account shall have an associated user profile that shows the quiz history, created quizzes, and points of the user.
- 1.1.1.7 The account's profile details shall be able to be changed with a profile details page.
</blockquote>
</details>
<details><summary>1.1.2. Registration and Login Requirements</summary><blockquote>
- 1.1.2.1 The user shall be able to register to the system with the following pieces of information: username, name, e-mail, password, and English proficiency level.
- 1.1.2.2 The account shall have passwords whose encrypted versions are securely stored in the system.
- 1.1.2.3 The system shall allow the user to change their passwords, either through the old password or through account recovery with e-mails.
- 1.1.2.4 The system shall allow the user to log in with their username/e-mail and password.
</blockquote>
</details>
<details><summary>1.1.3. Quiz Requirements</summary><blockquote>
- 1.1.3.1 Users shall be able to create their own quizzes and share them with others.
- 1.1.3.2 Users shall accumulate points by completing quizzes within given time intervals.
- 1.1.3.3 Users shall be able to solve quizzes created by other users.
</blockquote>
</details>
<details><summary>1.1.4. User Interaction Requirements</summary><blockquote>
- 1.1.4.1 Users shall be able to see other users' profiles, including their created quizzes and points.
- 1.1.4.2 Users shall be able to share their created quizzes with other users.
- 1.1.4.3 Users shall be able to view leaderboards or rankings based on accumulated points within their proficiency group.
</blockquote>
</details>
<details><summary>1.1.5. Forum Requirements</summary><blockquote>
- 1.1.5.1 Users shall be able to create forum posts to ask questions about English words or language learning.
- 1.1.5.2 Users shall be able to reply to forum posts and engage in discussions.
- 1.1.5.3 Users shall be required to link their questions to relevant entries in Lexvo.

</blockquote>
</details>

These were proposed to me by Fahreddin.

**Comment by arincdemir** [comment]
I changed the starts of the sentences to "The user shall be able to".
I removed 1.1.2.2 because this fits non-functional requirements better. It is not what can be done, it is how it should be done.
I seperated 1.1.1.6 into multiple points.
I added - 1.1.3.2 Users shall be able to add their liked questions to their quizzes.
I modified - 1.1.3.5 Users shall be able to view quizzes created by other users that fit their proficiency level. 

Now I'll start thinking about other stuff that we might need to add.
